modules/wpml/wpml-config.php

```diff
@@ -262,6 +286,101 @@ public function translate_taxonomies( $taxonomies, $hide ) {
 		return $taxonomies;
 	}
 
+	/**
+	 * Translation management for strings in blocks content.
+	 *
+	 * @since 3.3
+	 *
+	 * @param string[][] $parsing_rules Rules as Xpath expressions to evaluate in the blocks content.
+	 * @return string[][] Rules completed with ones from wpml-config file.
+	 *
+	 * @phpstan-param array<string,array<string>> $parsing_rules
+	 * @phpstan-return array<string,array<string>>
+	 */
+	public function translate_blocks( $parsing_rules ) {
+		return array_merge( $parsing_rules, $this->get_blocks_parsing_rules( 'xpath' ) );
+	}
+
+	/**
+	 * Translation management for blocks attributes.
+	 *
+	 * @since 3.3
+	 *
+	 * @param string[][] $parsing_rules Rules for blocks attributes to translate.
+	 * @return string[][] Rules completed with ones from wpml-config file.
+	 *
+	 * @phpstan-param array<string,array<string>> $parsing_rules
+	 * @phpstan-return array<string,array<string>>
+	 */
+	public function translate_blocks_attributes( $parsing_rules ) {
+		return array_merge( $parsing_rules, $this->get_blocks_parsing_rules( 'key' ) );
+	}
+
+	/**
+	 * Returns rules to extract a kind of string for blocks.
+	 *
+	 * @since 3.3
+	 *
+	 * @param string $rule_tag Tag name to extract.
+	 * @return string[][] Rules completed with ones from wpml-config file.
+	 */
+	protected function get_blocks_parsing_rules( $rule_tag ) {
+
+		if ( isset( $this->parsing_rules[ $rule_tag ] ) ) {
+			return $this->parsing_rules[ $rule_tag ];
+		}
+		$this->parsing_rules = $this->extract_blocks_parsing_rules();
+		return $this->parsing_rules[ $rule_tag ];
```

Comment:
I'm not sure `$this->parsing_rules[ $rule_tag ]` is always set when you return, so suggesting this:
```suggestion

		if ( null === $this->parsing_rules  ) {
			$this->parsing_rules = $this->extract_blocks_parsing_rules();
		}
		
		return isset( $this->parsing_rules[ $rule_tag ] ) ? $this->parsing_rules[ $rule_tag ] : array();
```

Reply:
`isset()` seems to return `false` when the variable has a `null` value
https://www.php.net/manual/en/function.isset.php